projects/storefrontlib/shared/components/chat-messaging/messaging/messaging.component.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  AfterViewInit,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { WindowRef } from '@spartacus/core';
+import { ICON_TYPE } from 'projects/storefrontlib/cms-components';
+import { FilesFormValidators } from 'projects/storefrontlib/shared/services';
+import { Observable } from 'rxjs';
+import { MessageDetails, MessagingConfigs } from './messaging.model';
+
+@Component({
+  selector: 'cx-messaging',
+  templateUrl: './messaging.component.html',
+})
+export class MessagingComponent implements OnInit, AfterViewInit {
+  @Input() messageDetails$: Observable<MessageDetails>;
+  @Input() scrollToInput?: boolean = true;
+  @Input() messagingConfigs?: MessagingConfigs;
+
+  @Output()
+  send = new EventEmitter<{ files: FileList | undefined; message: string }>();
+
+  iconTypes = ICON_TYPE;
+  form: FormGroup;
+
+  MAX_INPUT_CHARACTERS: number = 2000;
+  MAX_SIZE: number = 10;
+  MAX_ENTRIES: number = 1;
```

Comment:
nit:  Shouldn't these be taken from a constant file?
I am OK is these are hardcoded here

Reply:
We can not use the customer ticketing constant file. I decided to declare it here as it's not being used outside of this component. what do you suggest?

Reply:
I am Ok with it.  the hosting component can inject their values using the message config anyways 👍